vim/_hl_groups.colortemplate

```diff
@@ -104,10 +104,10 @@ NonText              none          none
 Question             none          none
 SignColumn           none          none
 SpecialKey           none          none
-SpellBad             none          none
-SpellCap             none          none
-SpellLocal           none          none
-SpellRare            none          none
+SpellBad             br_red        none          t=underline,reverse g=undercurl s=red
+SpellCap             br_red        none          t=underline,reverse g=undercurl s=red
+SpellLocal           yellow        none          t=underline,reverse g=undercurl s=red
+SpellRare            cyan          none          t=underline,reverse g=undercurl s=cyan
```

Comment:
Maybe a terminal issue? Have you tried with a different terminal?

Reply:
Yes, I've tried with Alacritty, and Gnome Terminal, kitty and a few others same result across them all.

Reply:
Perhaps, it is worth opening an issue at Vim's repo, or at least ask in Vim users' mailing list.

Reply:
@dngray I'd suggest styling other interface elements first and leave this for later, what do you think? In the meantime maybe we'll find someone who can help us with solving it.

Reply:
That is the plan

Reply:
Yes I haven't had a chance to post about this in the Vim communities, hopefully I will tonight. Maybe someone in /r/Vim or vim-subscribe@vim.org could help.

Reply:
The changes this generates:

```diff
@@ -125,10 +125,10 @@
     hi Question guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SignColumn guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SpecialKey guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellBad guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellCap guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellLocal guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellRare guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
+    hi SpellBad guifg=#ff665c guibg=NONE guisp=#fa5750 gui=reverse,undercurl cterm=reverse,undercurl
+    hi SpellCap guifg=#ff665c guibg=NONE guisp=#fa5750 gui=reverse,undercurl cterm=reverse,undercurl
+    hi SpellLocal guifg=#dbb32d guibg=NONE guisp=#fa5750 gui=reverse,undercurl cterm=reverse,undercurl
+    hi SpellRare guifg=#41c7b9 guibg=NONE guisp=#41c7b9 gui=reverse,undercurl cterm=reverse,undercurl
     hi Title guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
     hi VisualNOS guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
     hi WarningMsg guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
@@ -245,10 +245,10 @@
   hi Question guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SignColumn guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SpecialKey guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellBad guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellCap guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellLocal guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellRare guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
+  hi SpellBad guifg=#cc1729 guibg=NONE guisp=#d2212d gui=reverse,undercurl cterm=reverse,undercurl
+  hi SpellCap guifg=#cc1729 guibg=NONE guisp=#d2212d gui=reverse,undercurl cterm=reverse,undercurl
+  hi SpellLocal guifg=#ad8900 guibg=NONE guisp=#d2212d gui=reverse,undercurl cterm=reverse,undercurl
+  hi SpellRare guifg=#009c8f guibg=NONE guisp=#009c8f gui=reverse,undercurl cterm=reverse,undercurl
   hi Title guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
   hi VisualNOS guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
   hi WarningMsg guifg=NONE guibg=NONE guisp=NONE gui=NONE ctermfg=NONE ctermbg=NONE cterm=NONE
@@ -349,10 +349,10 @@
     hi Question ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SignColumn ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SpecialKey ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellBad ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellCap ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellRare ctermfg=NONE ctermbg=NONE cterm=NONE
+    hi SpellBad ctermfg=203 ctermbg=NONE cterm=reverse,underline
+    hi SpellCap ctermfg=203 ctermbg=NONE cterm=reverse,underline
+    hi SpellLocal ctermfg=178 ctermbg=NONE cterm=reverse,underline
+    hi SpellRare ctermfg=44 ctermbg=NONE cterm=reverse,underline
     hi Title ctermfg=NONE ctermbg=NONE cterm=NONE
     hi VisualNOS ctermfg=NONE ctermbg=NONE cterm=NONE
     hi WarningMsg ctermfg=NONE ctermbg=NONE cterm=NONE
@@ -448,10 +448,10 @@
   hi Question ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SignColumn ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SpecialKey ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellBad ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellCap ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellRare ctermfg=NONE ctermbg=NONE cterm=NONE
+  hi SpellBad ctermfg=160 ctermbg=NONE cterm=reverse,underline
+  hi SpellCap ctermfg=160 ctermbg=NONE cterm=reverse,underline
+  hi SpellLocal ctermfg=136 ctermbg=NONE cterm=reverse,underline
+  hi SpellRare ctermfg=37 ctermbg=NONE cterm=reverse,underline
   hi Title ctermfg=NONE ctermbg=NONE cterm=NONE
   hi VisualNOS ctermfg=NONE ctermbg=NONE cterm=NONE
   hi WarningMsg ctermfg=NONE ctermbg=NONE cterm=NONE
@@ -549,10 +549,10 @@
     hi Question ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SignColumn ctermfg=NONE ctermbg=NONE cterm=NONE
     hi SpecialKey ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellBad ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellCap ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=NONE
-    hi SpellRare ctermfg=NONE ctermbg=NONE cterm=NONE
+    hi SpellBad ctermfg=LightRed ctermbg=NONE cterm=reverse,underline
+    hi SpellCap ctermfg=LightRed ctermbg=NONE cterm=reverse,underline
+    hi SpellLocal ctermfg=DarkYellow ctermbg=NONE cterm=reverse,underline
+    hi SpellRare ctermfg=DarkCyan ctermbg=NONE cterm=reverse,underline
     hi Title ctermfg=NONE ctermbg=NONE cterm=NONE
     hi VisualNOS ctermfg=NONE ctermbg=NONE cterm=NONE
     hi WarningMsg ctermfg=NONE ctermbg=NONE cterm=NONE
@@ -648,10 +648,10 @@
   hi Question ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SignColumn ctermfg=NONE ctermbg=NONE cterm=NONE
   hi SpecialKey ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellBad ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellCap ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=NONE
-  hi SpellRare ctermfg=NONE ctermbg=NONE cterm=NONE
+  hi SpellBad ctermfg=LightRed ctermbg=NONE cterm=reverse,underline
+  hi SpellCap ctermfg=LightRed ctermbg=NONE cterm=reverse,underline
+  hi SpellLocal ctermfg=DarkYellow ctermbg=NONE cterm=reverse,underline
+  hi SpellRare ctermfg=DarkCyan ctermbg=NONE cterm=reverse,underline
   hi Title ctermfg=NONE ctermbg=NONE cterm=NONE
   hi VisualNOS ctermfg=NONE ctermbg=NONE cterm=NONE
   hi WarningMsg ctermfg=NONE ctermbg=NONE cterm=NONE
```

Reply:
https://groups.google.com/forum/?_escaped_fragment_=topic/vim_use/w4YKijB2zkY

Reply:
Why `reverse,underline'? The style for `Spell*` is supposed to be `undercurl`, which is rendered as a swiggly underline in GUI Vim and as an underline in TUI Vim.

Reply:
Looks more or less what [I suggested](https://github.com/jan-warchol/selenized/pull/71#issuecomment-607078816). Do you need any help, @dngray ?

Reply:
I haven't actually had time to do this, looking at having a go with it today.

Reply:
> Looks more or less what [I suggested](https://github.com/jan-warchol/selenized/pull/71#issuecomment-607078816). Do you need any help, @dngray ?

I tested this, seem to be getting a variety of results. Hmm.

![variety_of_results](https://user-images.githubusercontent.com/48640805/78417697-3814aa00-7624-11ea-8b22-6da139c012d7.png)

Reply:
So my guess is, not all terminals support undercurl, in the case of alacritty, it will fallback to _.

However I am not sure as to why it does not want to work in Vim with either Gnome Terminal or Alacritty.

Looking at the code colortemplate generates, we see in:



```vim
if (has('termguicolors') && &termguicolors) || has('gui_running')
    hi SpellBad guifg=NONE guibg=NONE guisp=#fa5750 gui=undercurl ctermfg=NONE ctermbg=NONE cterm=undercurl
    hi SpellCap guifg=NONE guibg=NONE guisp=#fa5750 gui=undercurl ctermfg=NONE ctermbg=NONE cterm=undercurl
    hi SpellLocal guifg=NONE guibg=NONE guisp=#dbb32d gui=undercurl ctermfg=NONE ctermbg=NONE cterm=undercurl
    hi SpellRare guifg=NONE guibg=NONE guisp=#41c7b9 gui=undercurl ctermfg=NONE ctermbg=NONE cterm=undercurl
```

```vim
if s:t_Co >= 256
    hi SpellBad ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellCap ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellRare ctermfg=NONE ctermbg=NONE cterm=underline
```

```vim
if s:t_Co >= 16
    hi SpellBad ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellCap ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellLocal ctermfg=NONE ctermbg=NONE cterm=underline
    hi SpellRare ctermfg=NONE ctermbg=NONE cterm=underline
```

Reply:
> Why `reverse,underline`? The style for `Spell*` is supposed to be `undercurl`, which is rendered as a swiggly underline in GUI Vim and as an underline in TUI.
> 
> `cterm=undercurl` is guaranteed to work in Vim. It is the right value to use.

I seem as above with `cterm=undercurl`.